### DISCUSSION TITLE
Add an `is_planar` function?

### DISCUSSION THREAD

**Discussion by rossbar** [networkx/networkx#5109] [Ideas]
This idea is inspired by #5079 and the related discussion in #5090

According to the above linked issue/discussion, the preferred way to check for planarity of a Graph is with the `nx.check_planarity` function; however, the above also suggest there are discoverability issues for this function. Documenting (e.g. fixing #5090) will help, but I wonder if the discoverability problem couldn't be further improved by adding an `is_planar` function.

The `nx.is_*` is a very common pattern in NetworkX for checking properties of Graphs:

```python
>>> import networkx as nx
>>> [func for func in dir(nx) if func.startswith('is_')]
['is_aperiodic',
 'is_arborescence',
 'is_at_free',
 'is_attracting_component',
 'is_biconnected',
 'is_bipartite',
 'is_branching',
 'is_chordal',
 'is_connected',
 'is_digraphical',
 'is_directed',
 'is_directed_acyclic_graph',
 'is_distance_regular',
 'is_dominating_set',
 'is_edge_cover',
 'is_empty',
 'is_eulerian',
 'is_forest',
 'is_frozen',
 'is_graphical',
 'is_isolate',
 'is_isomorphic',
 'is_k_edge_connected',
 'is_k_regular',
 'is_kl_connected',
 'is_matching',
 'is_maximal_matching',
 'is_multigraphical',
 'is_negatively_weighted',
 'is_path',
 'is_perfect_matching',
 'is_pseudographical',
 'is_regular',
 'is_semiconnected',
 'is_semieulerian',
 'is_simple_path',
 'is_strongly_connected',
 'is_strongly_regular',
 'is_tree',
 'is_triad',
 'is_valid_degree_sequence_erdos_gallai',
 'is_valid_degree_sequence_havel_hakimi',
 'is_valid_directed_joint_degree',
 'is_valid_joint_degree',
 'is_weakly_connected',
 'is_weighted']
```

As a naive NetworkX user, when I want to perform some sort of property check on a Graph, I typically find myself doing `nx.is_<tab>` in IPython and searching through options this way. Thus adding an `nx.is_planar` function would improve the discoverability of NetworkX's planarity-checking functionality.

One obvious downside is the increased API footprint and duplication of functionality: `is_planarity` would essentially be duplicating the functionality of `check_planarity` (minus the explicit search/returning of a counter-example). In this case, I think the benefits outweigh the drawbacks. Thoughts?

**Comment by dschult** [comment]
I like the function name idiom `is_*` and it might be that `check_planarity` could just be renamed `is_planar`. But to the extent that the `is_planar` could avoid finding the counter-example, maybe two functions with well connected docstrings could work well.  Both could be made more prominent in the planar_embedding documentation.

I wonder if any of the other properties has this trait that you want a function that reports `is_*`, yet you also want a function that provides information about that property.  Even if none of the others have this trait, I agree that the benefits of having 2 functions outweigh the drawbacks of a more crowded API.

**Comment by dtekinoglu** [comment]
Just opened a [PR](https://github.com/networkx/networkx/pull/5544) related to this issue !
